pkg/controller/volume/attachdetach/cache/desired_state_of_world.go

```diff
@@ -267,10 +274,36 @@ func (dsw *desiredStateOfWorld) DeleteNode(nodeName k8stypes.NodeName) error {
 	}
 
 	if len(nodeObj.volumesToAttach) > 0 {
+		// query real status of node
+		// if node is in safe mode mark as safeToDetach
+		safeToDetach := true
+		for _, volumeObj := range nodeObj.volumesToAttach {
```

Comment:
here I am looping all of the volumes in same node. However, I am thinking is that too much. Lets imagine that we have 10 volumes in same node - it will request same node status from cloudprovider api 10 times. Should we just take first volume and check status from that node?